core/domain/collection_domain.py

```
@@ -344,6 +345,93 @@ def from_dict(
 
         return collection
 
+    @classmethod
+    def deserialize(cls, memory_cache_json_string):
+        """Return a Collection domain object decoded from a JSON string
+        retrieved from the memory cache.
+
+        Args:
+            memory_cache_json_string: str. A JSON-encoded string that can be
+                decoded into a dictionary representing a Collection. Only call
+                on strings returned from caching_services.get_multi.
+
+        Raises:
+            Exception: The string is not a valid JSON string.
+
+        Returns:
+            Collection. The corresponding Collection domain object.
+        """
+        try:
+            collection_dict = json.loads(memory_cache_json_string)
+        except ValueError:
```

Why would this arise? Also the error message isn't helpful anyway, you won't know what the issue is by inspecting it.

---

If the string passed in is a random string that is not a json format, the json.loads function will crash and throw an exception. I can either handle it and make the error message more clear or just allow it to throw an exception on its own. Which do you think makes more sense here? I originally chose to use the error message to call to attention that the json string is trying to be decoded in a collection but I see now that wrapping around json.loads makes no sense since that decoding doesn't have anything to do with the Collection class.

---

Just let it throw an exception on its own. (This is what you should generally do for exceptions that you don't expect to happen in normal usage.)

---

Done!